float32 w_err
float32 h_err